### ISSUE TITLE
Midnight time steps partially missing in all 6hrLev files

### ISSUE THREAD

**Issue by plesager** [EC-Earth/ece2cmor3#653]
The problem reported here occurs with the CMIP/AerChemMIP/amip experiment. There are other experiments with 6hrLev output, although not AerChemMIP ones. I wonder if somebody else has tried and succeeded to output IFS variables on 6hrLev. In any case here's what I'm facing. 

`nctime` reports an error (incorrect time axis) in all cmorized IFS output in 6hrLev. TM5 output in 6hrLev is fine. The full error message is:

```hus_6hrLev_EC-Earth3-AerChem_amip_r1i1p1f1_gr_198201010000-198212311800.nc
        Units:
            IN FILE -- days since 1960-01-01 00:00:00
            REF     -- days since 1960-01-01 00:00:00
        Calendar:
            IN FILE -- proleptic_gregorian
            REF     -- proleptic_gregorian
        Start:
            IN FILE -- 198201010000 = 1982-01-01T00:00:00 = 8036.0
            REF     -- 198201010000 = 1982-01-01T00:00:00 = 8036.0
        End:
            FILENAME -- 198212311800 = 1982-12-31T18:00:00 = 8400.75
            IN FILE  -- 198212311800 = 1982-12-31T18:00:00 = 8400.75
            REBUILT  -- 198210090600 = 1982-10-09T06:00:00 = 8317.25
        Length: 1126
        MIP table: 6hrLev
        Frequency: 6hrPt = 6 hours
        Is instant: True
        Is climatology: False
        Has bounds: False
        Status: Error 001 -- Incorrect time axis over one or several time steps
        Status: Error 008b -- End date from filename is later than theoretical end date
        Status: Error 009b -- End date in file is later than theoretical end date
        Wrong time step at index 5: IN FILE -- 1982-01-02T06:00:00 = 8037.25    vs. REBUILT -- 1982-01-02T00:00:00 = 8037.0
        Wrong time step at index 6: IN FILE -- 1982-01-02T12:00:00 = 8037.5     vs. REBUILT -- 1982-01-02T06:00:00 = 8037.25
        Wrong time step at index 7: IN FILE -- 1982-01-02T18:00:00 = 8037.75    vs. REBUILT -- 1982-01-02T12:00:00 = 8037.5
        Wrong time step at index 8: IN FILE -- 1982-01-03T06:00:00 = 8038.25    vs. REBUILT -- 1982-01-02T18:00:00 = 8037.75
        Wrong time step at index 9: IN FILE -- 1982-01-03T12:00:00 = 8038.5     vs. REBUILT -- 1982-01-03T00:00:00 = 8038.0
```

I checked the cmorized files, and indeed something is wrong. `cdo` is not complaining but note how the **midnight steps** are missing at the beginning but not at the end:  

```  [5981] >>> cdo sinfo ./6hrLev/hus/gr/v20200901/hus_6hrLev_EC-Earth3-AerChem_amip_r1i1p1f1_gr_197901010000-197912311800.nc
   File format : NetCDF4 classic zip
    -1 : Institut Source   T Steptype Levels Num    Points Num Dtype : Parameter ID
     1 : unknown  EC-Earth3-AerChem v instant       1   1    131072   1  F32z : -1            
     2 : unknown  EC-Earth3-AerChem v instant      91   2    131072   1  F32z : -2            
   Grid coordinates :
     1 : gaussian                 : points=131072 (512x256)  F128
                              lon : 0 to 359.2969 by 0.703125 degrees_east  circular
                              lat : -89.46282 to 89.46282 degrees_north
                        available : cellbounds
   Vertical coordinates :
     1 : surface                  : levels=1
     2 : hybrid                   : levels=91
                              lev : 0.9988151 to 1.250025e-05 1
                           bounds : 1-0.9976302 to 2.50005e-05-0 1
                        available : vct  ps: ps
   Time coordinate :  1126 steps
     RefTime =  1960-01-01 00:00:00  Units = days  Calendar = proleptic_gregorian
  YYYY-MM-DD hh:mm:ss  YYYY-MM-DD hh:mm:ss  YYYY-MM-DD hh:mm:ss  YYYY-MM-DD hh:mm:ss
  1979-01-01 00:00:00  1979-01-01 06:00:00  1979-01-01 12:00:00  1979-01-01 18:00:00
  1979-01-02 06:00:00  1979-01-02 12:00:00  1979-01-02 18:00:00  1979-01-03 06:00:00
  1979-01-03 12:00:00  1979-01-03 18:00:00  1979-01-04 06:00:00  1979-01-04 12:00:00
  1979-01-04 18:00:00  1979-01-05 06:00:00  1979-01-05 12:00:00  1979-01-05 18:00:00
  1979-01-06 06:00:00  1979-01-06 12:00:00  1979-01-06 18:00:00  1979-01-07 06:00:00
  1979-01-07 12:00:00  1979-01-07 18:00:00  1979-01-08 06:00:00  1979-01-08 12:00:00
  1979-01-08 18:00:00  1979-01-09 06:00:00  1979-01-09 12:00:00  1979-01-09 18:00:00
  1979-01-10 06:00:00  1979-01-10 12:00:00  1979-01-10 18:00:00  1979-01-11 06:00:00
  1979-01-11 12:00:00  1979-01-11 18:00:00  1979-01-12 06:00:00  1979-01-12 12:00:00
  1979-01-12 18:00:00  1979-01-13 06:00:00  1979-01-13 12:00:00  1979-01-13 18:00:00
  1979-01-14 06:00:00  1979-01-14 12:00:00  1979-01-14 18:00:00  1979-01-15 06:00:00
  1979-01-15 12:00:00  1979-01-15 18:00:00  1979-01-16 06:00:00  1979-01-16 12:00:00
  1979-01-16 18:00:00  1979-01-17 06:00:00  1979-01-17 12:00:00  1979-01-17 18:00:00
  1979-01-18 06:00:00  1979-01-18 12:00:00  1979-01-18 18:00:00  1979-01-19 06:00:00
  1979-01-19 12:00:00  1979-01-19 18:00:00  1979-01-20 06:00:00  1979-01-20 12:00:00
   ................................................................................
   ................................................................................
   ..
  1979-12-17 12:00:00  1979-12-17 18:00:00  1979-12-18 00:00:00  1979-12-18 06:00:00
  1979-12-18 12:00:00  1979-12-18 18:00:00  1979-12-19 00:00:00  1979-12-19 06:00:00
  1979-12-19 12:00:00  1979-12-19 18:00:00  1979-12-20 00:00:00  1979-12-20 06:00:00
  1979-12-20 12:00:00  1979-12-20 18:00:00  1979-12-21 00:00:00  1979-12-21 06:00:00
  1979-12-21 12:00:00  1979-12-21 18:00:00  1979-12-22 00:00:00  1979-12-22 06:00:00
  1979-12-22 12:00:00  1979-12-22 18:00:00  1979-12-23 00:00:00  1979-12-23 06:00:00
  1979-12-23 12:00:00  1979-12-23 18:00:00  1979-12-24 00:00:00  1979-12-24 06:00:00
  1979-12-24 12:00:00  1979-12-24 18:00:00  1979-12-25 00:00:00  1979-12-25 06:00:00
  1979-12-25 12:00:00  1979-12-25 18:00:00  1979-12-26 00:00:00  1979-12-26 06:00:00
  1979-12-26 12:00:00  1979-12-26 18:00:00  1979-12-27 00:00:00  1979-12-27 06:00:00
  1979-12-27 12:00:00  1979-12-27 18:00:00  1979-12-28 00:00:00  1979-12-28 06:00:00
  1979-12-28 12:00:00  1979-12-28 18:00:00  1979-12-29 00:00:00  1979-12-29 06:00:00
  1979-12-29 12:00:00  1979-12-29 18:00:00  1979-12-30 00:00:00  1979-12-30 06:00:00
  1979-12-30 12:00:00  1979-12-30 18:00:00  1979-12-31 00:00:00  1979-12-31 06:00:00
  1979-12-31 12:00:00  1979-12-31 18:00:00
cdo    sinfo: Processed 2 variables over 1126 timesteps [0.56s 28MB].
```

Dumping the time variable shows that the midnight steps reappear from time 7276 on (29 days before end of the year):

``` time = 6940, 6940.25, 6940.5, 6940.75, 6941.25, 6941.5, 6941.75, 6942.25, 
    6942.5, 6942.75, 6943.25, 6943.5, 6943.75, 6944.25, 6944.5, 6944.75, 
    6945.25, 6945.5, 6945.75, 6946.25, 6946.5, 6946.75, 6947.25, 6947.5, 
    6947.75, 6948.25, 6948.5, 6948.75, 6949.25, 6949.5, 6949.75, 6950.25, 
    6950.5, 6950.75, 6951.25, 6951.5, 6951.75, 6952.25, 6952.5, 6952.75, 
    6953.25, 6953.5, 6953.75, 6954.25, 6954.5, 6954.75, 6955.25, 6955.5, 
    6955.75, 6956.25, 6956.5, 6956.75, 6957.25, 6957.5, 6957.75, 6958.25, 
    6958.5, 6958.75, 6959.25, 6959.5, 6959.75, 6960.25, 6960.5, 6960.75, 
    6961.25, 6961.5, 6961.75, 6962.25, 6962.5, 6962.75, 6963.25, 6963.5, 
    6963.75, 6964.25, 6964.5, 6964.75, 6965.25, 6965.5, 6965.75, 6966.25, 
   [...until...]
    7273.25, 7273.5, 7273.75, 7274.25, 7274.5, 7274.75, 7275, 7275.25, 
    7275.5, 7275.75, 7276, 7276.25, 7276.5, 7276.75, 7277, 7277.25, 7277.5, 
    7277.75, 7278, 7278.25, 7278.5, 7278.75, 7279, 7279.25, 7279.5, 7279.75, 
    7280, 7280.25, 7280.5, 7280.75, 7281, 7281.25, 7281.5, 7281.75, 7282, 
    7282.25, 7282.5, 7282.75, 7283, 7283.25, 7283.5, 7283.75, 7284, 7284.25, 
    7284.5, 7284.75, 7285, 7285.25, 7285.5, 7285.75, 7286, 7286.25, 7286.5, 
    7286.75, 7287, 7287.25, 7287.5, 7287.75, 7288, 7288.25, 7288.5, 7288.75, 
    7289, 7289.25, 7289.5, 7289.75, 7290, 7290.25, 7290.5, 7290.75, 7291, 
    7291.25, 7291.5, 7291.75, 7292, 7292.25, 7292.5, 7292.75, 7293, 7293.25, 
    7293.5, 7293.75, 7294, 7294.25, 7294.5, 7294.75, 7295, 7295.25, 7295.5, 
    7295.75, 7296, 7296.25, 7296.5, 7296.75, 7297, 7297.25, 7297.5, 7297.75, 
    7298, 7298.25, 7298.5, 7298.75, 7299, 7299.25, 7299.5, 7299.75, 7300, 
    7300.25, 7300.5, 7300.75, 7301, 7301.25, 7301.5, 7301.75, 7302, 7302.25, 
    7302.5, 7302.75, 7303, 7303.25, 7303.5, 7303.75, 7304, 7304.25, 7304.5, 
    7304.75 ;
```
I checked a raw output file and it looks fine. After isolating one variable, I got:

```[5966] >>> cdo sinfon filter.out 
   File format : GRIB
    -1 : Institut Source   T Steptype Levels Num    Points Num Dtype : Parameter name
     1 : ECMWF    unknown  v instant      39   1     65792   1  P16  : var130        
     2 : ECMWF    unknown  v instant      91   2     65792   1  P16  : var130        
   Grid coordinates :
     1 : spectral                 : points=65792  nsp=32896  truncation=255  complexPacking
   Vertical coordinates :
     1 : pressure                 : levels=39
                             plev : 100000 to 3 Pa
     2 : hybrid                   : levels=91
                              lev : 1 to 91 by 1 level
                        available : vct
   Time coordinate :  unlimited steps
     RefTime =  1960-05-01 06:00:00  Units = hours  Calendar = proleptic_gregorian
  YYYY-MM-DD hh:mm:ss  YYYY-MM-DD hh:mm:ss  YYYY-MM-DD hh:mm:ss  YYYY-MM-DD hh:mm:ss
  1960-05-01 06:00:00  1960-05-01 12:00:00  1960-05-01 18:00:00  1960-05-02 00:00:00
  1960-05-02 06:00:00  1960-05-02 12:00:00  1960-05-02 18:00:00  1960-05-03 00:00:00
  1960-05-03 06:00:00  1960-05-03 12:00:00  1960-05-03 18:00:00  1960-05-04 00:00:00
  1960-05-04 06:00:00  1960-05-04 12:00:00  1960-05-04 18:00:00  1960-05-05 00:00:00
  1960-05-05 06:00:00  1960-05-05 12:00:00  1960-05-05 18:00:00  1960-05-06 00:00:00
  1960-05-06 06:00:00  1960-05-06 12:00:00  1960-05-06 18:00:00  1960-05-07 00:00:00
  1960-05-07 06:00:00  1960-05-07 12:00:00  1960-05-07 18:00:00  1960-05-08 00:00:00
  1960-05-08 06:00:00  1960-05-08 12:00:00  1960-05-08 18:00:00  1960-05-09 00:00:00
  1960-05-09 06:00:00  1960-05-09 12:00:00  1960-05-09 18:00:00  1960-05-10 00:00:00
  1960-05-10 06:00:00  1960-05-10 12:00:00  1960-05-10 18:00:00  1960-05-11 00:00:00
  1960-05-11 06:00:00  1960-05-11 12:00:00  1960-05-11 18:00:00  1960-05-12 00:00:00
  1960-05-12 06:00:00  1960-05-12 12:00:00  1960-05-12 18:00:00  1960-05-13 00:00:00
  1960-05-13 06:00:00  1960-05-13 12:00:00  1960-05-13 18:00:00  1960-05-14 00:00:00
  1960-05-14 06:00:00  1960-05-14 12:00:00  1960-05-14 18:00:00  1960-05-15 00:00:00
  1960-05-15 06:00:00  1960-05-15 12:00:00  1960-05-15 18:00:00  1960-05-16 00:00:00
   ....
  1960-05-17 06:00:00  1960-05-17 12:00:00  1960-05-17 18:00:00  1960-05-18 00:00:00
  1960-05-18 06:00:00  1960-05-18 12:00:00  1960-05-18 18:00:00  1960-05-19 00:00:00
  1960-05-19 06:00:00  1960-05-19 12:00:00  1960-05-19 18:00:00  1960-05-20 00:00:00
  1960-05-20 06:00:00  1960-05-20 12:00:00  1960-05-20 18:00:00  1960-05-21 00:00:00
  1960-05-21 06:00:00  1960-05-21 12:00:00  1960-05-21 18:00:00  1960-05-22 00:00:00
  1960-05-22 06:00:00  1960-05-22 12:00:00  1960-05-22 18:00:00  1960-05-23 00:00:00
  1960-05-23 06:00:00  1960-05-23 12:00:00  1960-05-23 18:00:00  1960-05-24 00:00:00
  1960-05-24 06:00:00  1960-05-24 12:00:00  1960-05-24 18:00:00  1960-05-25 00:00:00
  1960-05-25 06:00:00  1960-05-25 12:00:00  1960-05-25 18:00:00  1960-05-26 00:00:00
  1960-05-26 06:00:00  1960-05-26 12:00:00  1960-05-26 18:00:00  1960-05-27 00:00:00
  1960-05-27 06:00:00  1960-05-27 12:00:00  1960-05-27 18:00:00  1960-05-28 00:00:00
  1960-05-28 06:00:00  1960-05-28 12:00:00  1960-05-28 18:00:00  1960-05-29 00:00:00
  1960-05-29 06:00:00  1960-05-29 12:00:00  1960-05-29 18:00:00  1960-05-30 00:00:00
  1960-05-30 06:00:00  1960-05-30 12:00:00  1960-05-30 18:00:00  1960-05-31 00:00:00
  1960-05-31 06:00:00  1960-05-31 12:00:00  1960-05-31 18:00:00  1960-06-01 00:00:00
cdo    sinfon: Processed 2 variables over 124 timesteps [1.33s 30MB].
```

Since I was using the `cosp-filter-fix` branch, I tried with the `master`. Same problem. Since that test was with another leg, I noticed that the midnight step starts reappearing 30 days before the end.

**Comment by goord** [comment]
Hi Philippe thanks for the reporting, I suspect my previous 'fast-forward' code to have something to do with this. Could you test this by changing line 534 of grib_filter.py to the following:
```python
proc_grib_file(grib_file.create_grib_file(fin), keys2files, grid, handles, once, None)
```
i.e. replace the ```cache``` argument with ```None```, which should disable the fast-forwarding.

**Comment by plesager** [comment]
That fixes it. Thanks.

**Comment by goord** [comment]
Ok I will fix this issue tomorrow

**Comment by goord** [comment]
Hi Philippe could you also do a small test with the filter-alevels-fix branch? I have pushed a fix there that doesn't disable the fast-forwarding in the grib files

**Comment by plesager** [comment]
Your `filter-alevels-fix` branch works, in the sense that the problem reported above disappears.

**Comment by goord** [comment]
Fixed in PR #658
